test/CMakeLists.txt

```diff
@@ -0,0 +1,74 @@
+#############################################################################
+# Copyright (c) 2023, xeus-cpp contributors                                 #
+#                                                                           #
+# Distributed under the terms of the BSD 3-Clause License.                  #
+#                                                                           #
+# The full license is in the file LICENSE, distributed with this software.  #
+#############################################################################
+
+
+# Unit tests
+# ==========
+
+cmake_minimum_required(VERSION 3.1)
+
+if (CMAKE_CURRENT_SOURCE_DIR STREQUAL CMAKE_SOURCE_DIR)
+    project(xeus-cpp-test)
+
+    enable_testing()
+
+    find_package(xeus-cpp REQUIRED CONFIG)
+endif()
+
+message(STATUS "Forcing tests build type to Release")
+set(CMAKE_BUILD_TYPE Release CACHE STRING "Choose the type of build." FORCE)
+
+include(CheckCXXCompilerFlag)
+
+string(TOUPPER "${CMAKE_BUILD_TYPE}" U_CMAKE_BUILD_TYPE)
+
+if(CMAKE_CXX_COMPILER_ID MATCHES Clang OR CMAKE_CXX_COMPILER_ID MATCHES GNU OR CMAKE_CXX_COMPILER_ID MATCHES Intel)
+    add_compile_options(-Wunused-parameter -Wextra -Wreorder -Wconversion -Wsign-conversion)
+
+    CHECK_CXX_COMPILER_FLAG(-march=native HAS_MARCH_NATIVE)
+    if (HAS_MARCH_NATIVE)
+        add_compile_options(-march=native)
+    endif()
+endif()
+
+if(CMAKE_CXX_COMPILER_ID MATCHES MSVC)
+    add_compile_options(/EHsc /MP /bigobj)
+    set(CMAKE_EXE_LINKER_FLAGS /MANIFEST:NO)
+endif()
+
+find_package(doctest)
```

Comment:
Let’s use gtest which seems standard for the projects that are close to llvm.

Reply:
I would prefer to keep doctest. gtest had a period with almost no maintainance (I had to keep a forked branch for more than one year before they merged and released a patch I submitted, and the bug prevented from testing xtensor ), and had a bunch of ABI compatibilities issues (we had to build it from source when trying to test with different compilers). Besides, doctest is used for the whole xeus stack now.